vendor/github.com/Microsoft/hcsshim/pkg/ociwclayer/cim/import.go

```diff
@@ -0,0 +1,161 @@
+//go:build windows
+// +build windows
+
+package cim
+
+import (
+	"archive/tar"
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/Microsoft/go-winio/backuptar"
+	"github.com/Microsoft/hcsshim/internal/wclayer/cim"
+	"github.com/Microsoft/hcsshim/pkg/ociwclayer"
+	"golang.org/x/sys/windows"
+)
+
+// ImportCimLayerFromTar reads a layer from an OCI layer tar stream and extracts it into
+// the CIM format at the specified path. The caller must specify the parent layers, if
+// any, ordered from lowest to highest layer.
+//
+// The caller must ensure that the thread or process has acquired backup and
+// restore privileges.
+//
+// This function returns the total size of the layer's files, in bytes.
+func ImportCimLayerFromTar(ctx context.Context, r io.Reader, layerPath string, parentLayerPaths []string) (int64, error) {
```

Comment:
I will add this on hcsshim, and once we have a final commit on hcsshim main that needs to vendored here, it will be updated.

Reply:
Are we still waiting for this?

Reply:
This change is in one of the pending PRs on hcsshim. We will have to vendor the latest commit of hcsshim once all CimFS PRs on hcsshim are merged. That should bring in the latest comment. You can see it here: https://github.com/ambarve/hcsshim/blob/cimfs_unionfs/pkg/ociwclayer/cim/import.go#L24

Reply:
Possibly too late to matter (I couldn't quickly find the PR for that branch), and only minor, but I clicked that link and the comment for `ImportCimLayerFromTar` made me think that literal `snapshots` was required in the directory path, but the implementation doesn't seem to require that. It's also not 100% explicit whether the `<id>` is an any way constrained or able to conflict with other things in the layers directory, e.g. a hypothetical snapshotter implementation that uses UUIDs or LCIDs instead of snapshot IDs for its directory names.